### PR TITLE
Allow add_dist_script() in subprojects

### PR DESCRIPTION
**Pull request by xclaesse**
Fixes: #8440.

### CODE REVIEW

**Comment by xclaesse** [comment]
@jpakkane @dcbaker any objection?

**Comment by dcbaker** [comment]
I didn't look over the implementation too closely, but I'm fine with idea. Consider this an acked-by

**Comment by jpakkane** [comment]
The thing that concerns me is that this may cause problems when people start disting subprojects that have not been updated to work with this. Dunno if we can do anything about it, though...

**Comment by xclaesse** [comment]
They certainly won't work out of the box, they have to be ported to use MESON_PROJECT_ variables otherwise all paths are going to be wrong. But since it's currently not allowed to call add_dist_script from a subproject, it can't create regressions.

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -1877,16 +1877,20 @@ the following methods.
...
   *(since 0.55.0)* The output of `configure_file`, `files`, and `find_program`
   as well as strings.
-
   *(since 0.57.0)* `file` objects and the output of `configure_file` may be
   used as the `script_name` parameter.
+  *(since 0.58.0)* This command can be invoked from a subproject, it was a hard
+  error in earlier versions. Subproject dist scripts will only be executed
+  when running `meson dist --include-subprojects`. `MESON_PROJECT_SOURCE_ROOT`
+  and `MESON_PROJECT_BUILD_ROOT` and `MESON_PROJECT_DIST_ROOT` environment
+  variables are set when dist scripts are run.
```
Comment (jpakkane) [inline docs/markdown/Reference-manual.md:1893]: This should specify what they are set to when running the master project. Perhaps also add some examples of the values one would expect from them.

Comment (xclaesse) [reply docs/markdown/Reference-manual.md]: Done. Also added the release notes file that I forgot to git add.

**Pull request merged**